src/ide.rs

```diff
@@ -642,8 +642,8 @@ pub fn ast(engine_state: &mut EngineState, file_path: &str) {
                 {
                     "type": "ast",
                     "span": {
-                        "start": span.start - offset,
-                        "end": span.end - offset,
+                        "start": span.start.checked_sub(offset),
+                        "end": span.end.checked_sub(offset),
```

Comment:
Should this use `saturating_sub`?

Reply:
Sometimes, the span `(0, 0)` is used when we encounter a parse error. So, I think we should use `saturating_sub` to match this (it will return 0 on negative overflow). `checked_sub` will return `None`/`null` instead, which I guess is more correct, but we already have the convention that the `(0, 0)` span is special (e.g., `Span::unknown`).